ui/src/Queues/QueueRowDetails.tsx

```diff
@@ -5,49 +5,142 @@ import Box from "@material-ui/core/Box";
 import Typography from "@material-ui/core/Typography";
 import Table from "@material-ui/core/Table";
 import TableHead from "@material-ui/core/TableHead";
-import {TableBody} from "@material-ui/core";
-import React, {useState} from "react";
-import QueueService from "../services/QueueService";
+import { TableBody, Tabs, Tab } from "@material-ui/core";
+import React, { useState } from "react";
+import { getQueueAttributes } from "../services/QueueService";
+import QueueMessagesList from "./QueueMessagesList";
+import { QueueMessagesData } from "./QueueMessageData";
 
-const RowDetails: React.FC<{ props: { isExpanded: boolean, queueName: string } }> = ({props}) => {
+const RowDetails: React.FC<{
+  isExpanded: boolean;
+  queueName: string;
+  queueData: QueueMessagesData;
+  fetchQueueMessages: (queueName: string) => Promise<void>;
+  deleteMessage: (
+    queueName: string,
+    messageId: string,
+    receiptHandle: string
+  ) => Promise<void>;
+  updateMessageExpandedState: (
+    queueName: string,
+    messageId: string | null
+  ) => void;
+}> = ({
+  isExpanded,
+  queueName,
+  queueData,
+  fetchQueueMessages,
+  deleteMessage,
+  updateMessageExpandedState,
+}) => {
+  const [attributes, setAttributes] = useState<Array<Array<string>>>([]);
+  const [activeTab, setActiveTab] = useState(0);
 
-    const [attributes, setAttributes] = useState<Array<Array<string>>>([]);
+  async function fetchQueueAttributes() {
+    const attributes = await getQueueAttributes(queueName);
+    setAttributes(attributes);
+  }
 
-    function getQueueAttributes() {
-        QueueService.getQueueAttributes(props.queueName).then(attributes => setAttributes(attributes))
-    }
+  const handleTabChange = (_event: React.ChangeEvent<{}>, newValue: number) => {
+    setActiveTab(newValue);
+  };
+
+  interface TabPanelProps {
+    children?: React.ReactNode;
+    index: number;
+    value: number;
+  }
+
+  function TabPanel(props: TabPanelProps) {
```

Comment:
Could you extract this component (together with the props type) outside of the `RowDetails`? Otherwise, it gets re-declared on every `RowDetails` render.